pkg/api_builder/lib/_client_utils.dart

```diff
@@ -0,0 +1,93 @@
+/// Utilities for use in generated client code.
+library client_utils;
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:meta/meta.dart';
+import 'package:http/http.dart' as http;
+
+class RequestException implements Exception {
+  final int status;
+  final Map<String, String> headers;
+  final List<int> body;
+
+  RequestException(this.status, this.headers, [this.body]);
+
+  @override
+  String toString() => 'RequestException(status = $status)';
+}
+
+class Client {
+  final String _baseUrl;
+  final http.Client _client;
+
+  Client(String baseUrl, {http.Client client})
+      : _baseUrl = baseUrl,
+        _client = client {
+    ArgumentError.checkNotNull(_baseUrl, 'baseUrl');
+  }
+
+  Future<T> _withClient<T>(FutureOr<T> Function(http.Client) fn) async {
```

Comment:
Nit: This won't reuse HTTP connections. I guess it won't be too hard to extend it with "like reuse for 1000 requests".

Reply:
It will if you specify `client` in the constructor.

Reply:
Ah, true, my mistake.